src/runtime/io.cpp

```diff
@@ -788,6 +788,9 @@ extern "C" LEAN_EXPORT obj_res lean_io_rename(b_obj_arg from, b_obj_arg to, lean
     // so we have to call the underlying windows API directly to get behavior consistent
     // with the unix-like OSs
     bool ok = MoveFileEx(string_cstr(from), string_cstr(to), MOVEFILE_REPLACE_EXISTING) != 0;
+    if (!ok) {
+        errno = GetLastError();
```

Comment:
Actually, I think some conversion is probably needed here...

Reply:
See other uses of `GetLastError`, we usually use the string overload of `io_result_mk_error` for lack of a conversion function

Reply:
In the longer run perhaps it's worth copying [`errmap` from CPython](https://github.com/python/cpython/blob/main/PC/errmap.h#L2)?